pytests/tests/test_enums.py

```diff
@@ -0,0 +1,179 @@
+import pytest
+import sys
+from pyo3_pytests import enums
+
+
+def test_complex_enum_variant_constructors():
+    int_variant = enums.ComplexEnum.Int(42)
+    assert isinstance(int_variant, enums.ComplexEnum.Int)
+
+    float_variant = enums.ComplexEnum.Float(3.14)
+    assert isinstance(float_variant, enums.ComplexEnum.Float)
+
+    str_variant = enums.ComplexEnum.Str("hello")
+    assert isinstance(str_variant, enums.ComplexEnum.Str)
+
+    empty_struct_variant = enums.ComplexEnum.EmptyStruct()
+    assert isinstance(empty_struct_variant, enums.ComplexEnum.EmptyStruct)
+
+    multi_field_struct_variant = enums.ComplexEnum.MultiFieldStruct(42, 3.14, True)
+    assert isinstance(multi_field_struct_variant, enums.ComplexEnum.MultiFieldStruct)
+
+
+@pytest.mark.parametrize(
+    "variant",
+    [
+        enums.ComplexEnum.Int(42),
+        enums.ComplexEnum.Float(3.14),
+        enums.ComplexEnum.Str("hello"),
+        enums.ComplexEnum.EmptyStruct(),
+        enums.ComplexEnum.MultiFieldStruct(42, 3.14, True),
+    ],
+)
+def test_complex_enum_variant_subclasses(variant: enums.ComplexEnum):
+    assert isinstance(variant, enums.ComplexEnum)
+
+
+def test_complex_enum_field_getters():
+    int_variant = enums.ComplexEnum.Int(42)
+    assert int_variant.i == 42
+
+    float_variant = enums.ComplexEnum.Float(3.14)
+    assert float_variant.f == 3.14
+
+    str_variant = enums.ComplexEnum.Str("hello")
+    assert str_variant.s == "hello"
+
+    multi_field_struct_variant = enums.ComplexEnum.MultiFieldStruct(42, 3.14, True)
+    assert multi_field_struct_variant.a == 42
+    assert multi_field_struct_variant.b == 3.14
+    assert multi_field_struct_variant.c is True
+
+
+# NOTE(mkovaxx): Same as the test below, but for Python < 3.10
+@pytest.mark.parametrize(
+    "variant",
+    [
+        enums.ComplexEnum.Int(42),
+        enums.ComplexEnum.Float(3.14),
+        enums.ComplexEnum.Str("hello"),
+        enums.ComplexEnum.EmptyStruct(),
+        enums.ComplexEnum.MultiFieldStruct(42, 3.14, True),
+    ],
+)
+def test_complex_enum_desugared_match(variant: enums.ComplexEnum):
```

Comment:
I've added variants of the test cases that "desugar" the match statement syntax for Python < 3.10.